edit-client/src/client.rs

```diff
@@ -321,13 +322,18 @@ pub trait ClientImpl {
     where
         Self: Sized,
     {
-        self.send_client(&UserToFrontendCommand::Controls(Controls{
+        let mut controls_object = &UserToFrontendCommand::Controls(Controls{
             keys: key_handlers::<Self>()
                 .into_iter()
                 .map(|x| (x.0, x.1, x.2))
                 .collect(),
             buttons: button_handlers::<Self>(state).1
-        })).expect("Could not send initial state");
+        });
+
+        if Some(controls_object) != self.state().last_controls {
```

Comment:
Hey @tcr. Have done most of the work; the only thing left to do now is fixing the error that occurs when comparing the `controls_object` and `last_controls`.